metrics/internal/collectors/registry.go

```diff
@@ -26,15 +26,18 @@ func RegisterCustomResourceCollectors(registry *prometheus.Registry, opts *optio
 	cephBlockPoolCollector := NewCephBlockPoolCollector(opts)
 	cephClusterCollector := NewCephClusterCollector(opts)
 	OBMetricsCollector := NewObjectBucketCollector(opts)
+	clusterAdvanceFeatureCollector := NewClusterAdvanceFeatureCollector(opts)
```

Comment:
This function could return nil, which would cause an issue at line 34. We should handle that.

Reply:
Noticed that other collector functions are also doing the same, so made another commit to handle all of it.
Let me know if you want me to restrict this only to the current new (AdvancedFeatureCollector) Object only (as a simple change).

Reply:
Added the changes in a simple `if` condition